mailpoet/lib/EmailEditor/Engine/SettingsController.php

```diff
@@ -191,12 +191,17 @@ public function getTheme(): \WP_Theme_JSON {
   }
 
   public function getStylesheetForRendering(): string {
-    $settings = $this->getTheme()->get_settings();
+    $coreThemeSettings = \WP_Theme_JSON_Resolver::get_core_data()->get_settings();
+    $emailThemeSettings = $this->getTheme()->get_settings();
     $css = '';
     // Font family classes
-    foreach ($settings['typography']['fontFamilies']['theme'] as $fontFamily) {
+    foreach ($emailThemeSettings['typography']['fontFamilies']['theme'] as $fontFamily) {
       $css .= ".has-{$fontFamily['slug']}-font-family { font-family: {$fontFamily['fontFamily']}; } \n";
     }
+    // Font size classes
```

Comment:
> This is needed for cases where we don't touch the HTML output
to insert font size from the preprocessor (e.g., list item)

Your comment in the commit message mentions that this change is needed for the original HTML.
Do you think that we could use those classes in our updated/generated HTML instead of inlining extracted values?
It's only an idea and question because I'm a bit curious here.

Reply:
> Do you think that we could use those classes in our updated/generated HTML instead of inlining extracted values?

@lysyjan  This is a really good question. 👍 

I considered this option, but in the end, I think it was not worth it. For two reasons:

1) We allow custom font size values, we need to render those as CSS rules anyway. With classes we would still keep support also for pixel values. Supporting both classes and values at the block level is more complex than just values.
2) Because we reset the font size to 0 at a couple of wrappers, we need to bubble the values down. For custom font-size we send pixel value, if we wanted to support also classes we would have to bubble down both. So, supporting both classes and values at the block level is more complex than just values.

So this inlining currently covers only list items blocks. In general this could cover all blocks which are leaf blocks and which keep the original HTML from Gutenberg.

Does that make sense to you?